### PR TITLE
Prevent multi-line chat messages server-side

### PR DESCRIPTION
**Pull request by rubenwardy**
@luk3yx

### CODE REVIEW

**Comment by rubenwardy** [comment]
trim() function needs checking to make sure that it's correct

File: src/server.cpp [hunk 1]
```diff
@@ -2874,8 +2874,14 @@ std::wstring Server::handleChat(const std::string &name, const std::wstring &wna
 				L"It was refused. Send a shorter message";
 	}
 
+	auto message = wide_to_utf8(wmessage);
+	auto first = message.find_first_of("\n\r");
```
Comment (nerzhul) [inline src/server.cpp]: \r is not used on unix, just check \n

Comment (rubenwardy) [reply src/server.cpp]: it is on Windows, though.

Will logging to file strip away /r?

Comment (luk3yx) [reply src/server.cpp]: `\r` could produce unexpected results when using `cat` or `tail`:

```
$ echo $'line1\rline2\nline3'
line2
line3
```

**Comment by paramat** [comment]
What issue does this attend to and why is it a blocker? No description, no links in first post.

File: src/server.cpp (hunk 1 above)
Comment (ClobberXD) [reply src/server.cpp]: Maybe check for `\n` _or_ `\r`?

Comment (rubenwardy) [reply src/server.cpp]: That's what this does

File: src/server.cpp [hunk 2]
```diff
@@ -2874,8 +2874,14 @@ std::wstring Server::handleChat(const std::string &name, const std::wstring &wna
...
+	if (first != std::wstring::npos && !trim(message.substr(first)).empty()) {
```
Comment (SmallJoker) [inline src/server.cpp]: Neither `\n` nor `\r` are sent in regular chat messages. Thus latter check is not just superfluous, but would also allow players to add empty lines in the chat (after their message).

Comment (rubenwardy) [reply src/server.cpp]: The latter check is for trailing new lines. It could be trimmed instead

Comment (SmallJoker) [reply src/server.cpp]: The trim function does not modify `message`. Please correct me when I missed something but a message like `hello world\n` would pass this check and be sent to the other players, including the additional new line.

**Comment by rubenwardy** [comment]
Done

File: src/server.cpp [hunk 3]
```diff
@@ -2874,8 +2874,14 @@ std::wstring Server::handleChat(const std::string &name, const std::wstring &wna
 				L"It was refused. Send a shorter message";
 	}
 
+	auto message = trim(wide_to_utf8(wmessage));
+	auto first = message.find_first_of("\n\r");
```
Comment (sfan5) [inline src/server.cpp]: this function call can (and should IMO) be moved inside the `if` condition

**Pull request merged**